### DISCUSSION TITLE
Is destroy() intended to be used?

### DISCUSSION THREAD

**Discussion by michaelperel** [alpinejs/alpine#3354] [1. Help]
I have been using the destroy() method, but am I worried that I should not use it because it is undocumented. Is this a private feature, or something that I should not rely on? It is extremely useful for cleaning up 3rd party resources like bootstrap modals to avoid memory leaks.

Thanks!

**Comment by imacrayon** [comment]
There’s been a few PRs related to `destroy` and this comment explains why it’s undocumented: https://github.com/alpinejs/alpine/pull/2717#issuecomment-1325719733

I might treat it like a “beta” feature for now, I think the maintainers recognize that there are valid use cases for destroy functionality though.
